### PR TITLE
Add configuration to filter ort packages for exclusion

### PR DESCRIPTION
**Pull request by blaumeiser-at-bosch**
Signed-off-by: Lars Geyer-Blaumeiser <lars.geyer-blaumeiser@bosch.io>

### Request Reviewer
@bs-ondem @sschuberth 

### Type of Change
*Type of change*:  improvement

### How Has This Been Tested?
Regression Test

### Checklist
Must:
- [x] All related issues are referenced in commit messages

### CODE REVIEW

File: modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt [hunk 1]
```diff
@@ -29,6 +29,9 @@ class OrtResultAnalyzer : ManualAnalyzer() {
         private val LOGGER = LoggerFactory.getLogger(OrtResultAnalyzer::class.java)
     }
 
+    private val filterExcludedArtifactsProperty = "filter.excluded"
```
Comment (sschuberth) [inline modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt:32]: I'm always a bit confused about whether "to filter" means "to keep something" or "to get rid of something". If we follow the logic of [Kotlin's filter](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/filter.html), it "returns a list containing only elements matching the given predicate", so here "to filter" means "to keep the filtrate". Further following that logic means that setting `filter.excluded = true` should *keep* (only) the excludes, which is not what we want here.

TL;DR, I propose to rename this property to "omit.excludes" to 1) avoid this confusion and 2) align with ORT's naming of the parameter. Further, I'd argue we could / should set the default to `true`, because in the context of creating reports, I'd say you always want to omit excluded packages.

Comment (blaumeiser-at-bosch) [reply modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt]: Good point, followed your proposal in the new commit

File: modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt [hunk 2]
```diff
@@ -29,6 +29,9 @@ class OrtResultAnalyzer : ManualAnalyzer() {
         private val LOGGER = LoggerFactory.getLogger(OrtResultAnalyzer::class.java)
     }
 
+    private val omitExcludedArtifactsProperty = "omit.excluded"
+    private var omitExcludedArtifacts = false
```
Comment (sschuberth) [inline modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt:33]: Didn't we also agree to set this default to `true`, or did I misunderstand something?

File: modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt [hunk 3]
```diff
@@ -37,6 +40,11 @@ class OrtResultAnalyzer : ManualAnalyzer() {
 
     override fun getName() = "OrtResult"
 
+    override fun configure(configMap: MutableMap<String, String>?) {
+        super.configure(configMap)
+        omitExcludedArtifacts = getBooleanConfigValue(omitExcludedArtifactsProperty, configMap)
```
Comment (sschuberth) [inline modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt:45]: Hmm, it just occurred to me that `getBooleanConfigValue()` seem to hard-code the default value to `false`, so it seems we *cannot* have a default of `true` for the property, as this function always gets called (?) and therefore if does not matter which default value we assign to `omitExcludedArtifacts` above...

File: modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt (hunk 2 above)
Comment (sschuberth) [reply modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt:33]: Also, if `configure()` is guranteed to *always* be called, and always *before* `createArtifactList()`, then we should make this a `private lateinit val omitExcludedArtifacts: Boolean`.

File: modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt [hunk 4]
```diff
@@ -37,6 +40,11 @@ class OrtResultAnalyzer : ManualAnalyzer() {
 
     override fun getName() = "OrtResult"
 
+    override fun configure(configMap: MutableMap<String, String>?) {
```
Comment (sschuberth) [inline modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt:43]: The signature should ideally be `configMap: Map<String, String>` if `configMap` is never written to and is guaranteed to never be `null`.

File: modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt (hunk 3 above)
Comment (sschuberth) [reply modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt:45]: We can probably fix this by doing:

    omitExcludedArtifacts = getConfigValue(omitExcludedArtifactsProperty, configMap, "true").toBoolean()

File: modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt (hunk 2 above)
Comment (blaumeiser-at-bosch) [reply modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt:33]: I reverse the logic, so that you have to configure this, if you do not want the scope reduced

File: modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt (hunk 3 above)
Comment (blaumeiser-at-bosch) [reply modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt:45]: See above

File: modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt [hunk 5]
```diff
@@ -29,6 +29,9 @@ class OrtResultAnalyzer : ManualAnalyzer() {
         private val LOGGER = LoggerFactory.getLogger(OrtResultAnalyzer::class.java)
     }
 
+    private val disableScopeReductionProperty = "disable.scope.reduction"
```
Comment (sschuberth) [inline modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt:32]: So, if you want to invert the logic here, how about simply calling this "keep excluded artifacts"?

File: modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt [hunk 6]
```diff
@@ -37,6 +40,11 @@ class OrtResultAnalyzer : ManualAnalyzer() {
 
     override fun getName() = "OrtResult"
 
+    override fun configure(configMap: Map<String, String>?) {
```
Comment (sschuberth) [inline modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt:43]: I still believe we can safely make this a non-nullable type (as I already did [here](https://github.com/eclipse/antenna/blob/df5d05ad192e19415dc6e00a99c940439ae1b784/modules/ort/src/main/kotlin/workflow/processors/enricher/OrtDownloaderProcessor.kt#L30); the fix to also make the map immutable is [here](https://github.com/eclipse/antenna/pull/554)) as in all places I've seen so far an empty map is passed instead of `null`.

File: modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt [hunk 7]
```diff
@@ -37,6 +40,11 @@ class OrtResultAnalyzer : ManualAnalyzer() {
...
+        super.configure(configMap)
+        omitExcludedArtifacts = ! getBooleanConfigValue(keepExcludedArtifactsProperty, configMap)
```
Comment (sschuberth) [inline modules/ort/src/main/kotlin/workflow/analyzers/OrtResultAnalyzer.kt:45]: Nit: I'm not sure about the style for Antenna, but I'd expect to be no space after `!`.
